include/deal.II/base/exceptions.h

```diff
@@ -668,7 +672,16 @@ namespace StandardExceptions
   /**
    * An error occurred reading or writing a file.
    */
-  DeclException0 (ExcIO);
+  DeclExceptionMsg (ExcIO,
+                    "An input/output error has occurred. Ther are a number of "
+                    "reasons why this may be happening. First, you may be "
+                    "trying to read from a file that doesn't exist or that is "
+                    "not readable given its file permissions. Second, you "
+                    "may be trying to write to a file to which file or "
+                    "directory permissions do not allow you to write. "
```

Comment:
I think we could be a bit more specific here. For me, I almost always get this exception when I set an output directory that does not exist, e.g. when one selects a subdirectory for data output but forgot to create the directory before running the program.